### ISSUE TITLE
Builds on OS X with osx_image take over Linux

### ISSUE THREAD

**Issue by eregon** [travis-ci/travis-ci#3975]
Hello,

We just started trying to run [RubySpec](https://github.com/ruby/rubyspec) on Travis with OS X.

@BanzaiMan gave us the tip of using `osx_image: beta-xcode6.3` in the `.travis.yml`.

However, after this change, it seems *all* builds use OS X and so the tests are not run anymore on Linux!
See https://travis-ci.org/ruby/rubyspec/jobs/64411601 for instance which should be a Linux build but the ruby version is `ruby 2.0.0p643 (2015-02-25 revision 49749) [x86_64-darwin14.1.0]` (**darwin**).

---

The original failure is at https://travis-ci.org/ruby/rubyspec/builds/64402829, from the `ruby -v` it is `darwin13.4` so Mavericks, right? This should be enough to run RubySpec.

So I tried to specify a more precise version, following http://rubies.travis-ci.org/, and the build is https://travis-ci.org/ruby/rubyspec/jobs/64412214.
It seems to use `ruby 2.2.0dev (2014-12-03 trunk 48697) [x86_64-darwin13]` (reading from the full log) which is too old for our purposes.
`ruby-head` seems to actually also resolves to the same version, which might be due to a [too old rvm](https://twitter.com/hiro_asari/status/603906184073064449)?

In conclusion, it seems that version of OS X/rvm is too old.

---

In any case, thank you for allowing RubySpec to run on OS X on Travis CI and let's try to fix the bugs!

**Comment by BanzaiMan** [comment]
My initial guess is that the `osx_image` flag is evaluated first and sends the job to the OS X queue, regardless of the rest of the configuration.

**Comment by hsbt** [comment]
Hi, our configuration is here: https://github.com/ruby/rubyspec/commit/82b3c0a20125b2fb8b6ab989c87d95e8c0b084d4

**Comment by thomasvl** [comment]
Any update/progress on this?  With the default being such an old Xcode it makes multi os support hard to test.

**Comment by thomasvl** [comment]
Actually, did this get fixed without being marked as such?  Trying a test out, it seems like it might have been.

**Comment by ejona86** [comment]
I'm using matrix for other reasons, but earlier I did one build just to test if it was still an issue and didn't have any problems. I would believe it is fixed.

**Comment by eregon** [comment]
I'll close this then, thanks for your reports.

**Comment by thomasvl** [comment]
fyi - opened issue 5125 for the UI issue where it reports the Xcode version even for linux builds.

**Comment by graingert** [comment]
#5125

**Comment by graingert** [comment]
>  graingert unassigned sarahhodne just now 

how did I do that?

**Comment by gdevenyi** [comment]
This is still an issue. Instead of OSX taking over Linux, now multiple osx_image versions multiply the Linux build in the matrix
